seismicpro/src/seismic_batch.py

```diff
@@ -895,6 +896,196 @@ def equalize(self, index, src, dst, params, survey_id_col=None, upscale=False):
+        dst : str
+            The batch component to put semblance in.
+        velocities : array-like
+            A list of velocities to calculate semblance for.
+        win_size : int, optional, by default 25 samples.
+            Window size for smoothing semblance over the time axis.
+            Measured in samples.
+
+        Returns
+        -------
+        batch : SeismicBatch
+            Batch with calculated vertical velocity semblance in `dst` component.
+            `dst` elements are now instances of :class:`~semblance.Semblance` class.
+
+        Raises
+        ------
+        ValueError
+            If seismogram is not sorted by `offset`.
+        """
+        times = self.meta[src]['samples']
+
+        if self.meta[src]['sorting'] != 'offset':
+            raise ValueError('Seismogram should be sorted by `offset`.')
+
+        self._calculate_semblance(src=src, dst=dst, times=times, velocities=velocities, win_size=win_size)
+        return self
+
+    @inbatch_parallel(init="_init_component", target='threads')
+    def _calculate_semblance(self, index, src, dst, times, velocities, win_size):
+        pos = self.index.get_pos(index)
+        seismogram = getattr(self, src)[pos]
+        offsets = np.sort(self.index.get_df(index=index)['offset'])
+
+        semblance = Semblance(seismogram=seismogram, times=times, offsets=offsets,
+                              velocities=velocities, win_size=win_size)
+        getattr(self, dst)[pos] = semblance
+
+    @action
+    def calculate_residual_semblance(self, src, dst, num_vels, stacking_velocities, win_size=25, relative_margin=0.2):
+        """ Calculate the vertical residual semblance for a given seismogram from `src` component and save the result
+        to `dst` component.
+
+        See detailed documentation in :class:`~semblance.ResidualSemblance`.
+
+        Parameters
+        ----------
+        src : str
+            The batch component to get the seismogram from.
+        dst : str
+            The batch component to put semblance in.
+        num_vels : array-like
+            The number of velocities that are involved in the calculation of the vertical residual semblance.
+        stacking_velocities : array-like or str, optional
+            If array, is an array with elements in format [[time, velocity], ...]. Time measured in milliseconds,
+            velocity in meters/second. The array contains a set of non-decreasing velocity points that correspond
+            to the maximum correlation values.
+            Else is a component's name with stacking_velocities.
+        win_size : int, optional, by default 25
+            Window size for smoothing semblance over the time axis.
+            Measured in samples.
+        relative_margin : float, optional, by default 0.2
+            The relative velocity margin from the stacking velocity. Defines the width of the area around the
+            stacking velocity to construct the residual semblance.
```

Comment:
```suggestion
            Defines a relative velocity margin from the stacking velocity to construct the residual semblance.
```
Sounds more concise.